src/orbits/keplerian.jl

```diff
@@ -0,0 +1,532 @@
+using AstroLib: trueanom, kepler_solver
+using PhysicalConstants
+using Unitful, UnitfulAstro
+using KeywordCalls
+using Rotations
+
+# Domain specific unit conversions / Constants
+const G_nom = 2942.2062175044193 # Rsun^3/Msun/d^2
+const G_unit = PhysicalConstants.CODATA2018.G
+const MsunRsun_to_gcc = (1.0u"Msun/Rsun^3" |> u"g/cm^3").val
+
+"""
+    KeplerianOrbit(; kwargs...)
+
+Keplerian orbit parameterized by the basic observables of a transiting 2-body system.
+
+# Parameters
+* `period`/`P` -- The orbital period of the planet [d].
+* `t_0`/`t₀` -- The midpoint time of the reference transit [d].
+* `t_p`/`tₚ` -- The time of periastron [d].
+* `duration`/`τ` -- The transit duration [d].
+* `a` -- The semi-major axis [R⊙].
+* `R_planet`/`Rₚ` -- The radius of the planet [R⊙].
+* `R_star`/`Rₛ` -- The radius of the star [R⊙].
+* `rho_star`/`ρₛ` -- The spherical star density [M⊙/R⊙³].
+* `RpRs` -- The ratio of the planet radius to star radius.
+* `aR_star`/`aRₛ` -- The ratio of the semi-major axis to star radius.
+* `b` -- The impact parameter, bounded between 0 ≤ b ≤ 1.
+* `ecc` -- The eccentricity of the closed orbit, bounded between 0 ≤ ecc < 1.
+* `cos_omega`/`cos_ω` -- The cosine of the argument of periapsis.
+* `sin_omega`/`sin_ω` -- The sine of the argument of periapsis.
+* `incl` -- The inclination of the orbital plane relative to the axis perpendicular to the
+           reference plane [rad]
+* `omega`/`ω` -- The argument of periapsis [rad].
+* `Omega`/`Ω` -- The longitude of the ascending node [rad].
+* `M_planet`/`Mₚ` -- The mass of the planet [M⊙].
+* `M_star`/`Mₛ` -- The mass of the star [M⊙].
+
```

Comment:
okay, I think the final parameter-name options should look like the following:
```suggestion
# Parameters
* `period`/`P` -- The orbital period of the planet [d].
* `t0`/`t_0` -- The midpoint time of the reference transit [d].
* `tp`/`t_p` -- The time of periastron [d].
* `duration`/`τ` -- The transit duration [d].
* `a` -- The semi-major axis [R⊙].
* `R_planet`/`Rp` -- The radius of the planet [R⊙].
* `R_star`/`Rs` -- The radius of the star [R⊙].
* `rho_star`/`ρ_star` -- The spherical star density [M⊙/R⊙³].
* `RpRs` -- The ratio of the planet radius to star radius.
* `aR_star`/`aRs` -- The ratio of the semi-major axis to star radius.
* `b` -- The impact parameter, bounded between 0 ≤ b ≤ 1.
* `ecc` -- The eccentricity of the closed orbit, bounded between 0 ≤ ecc < 1.
* `cos_omega`/`cos_ω` -- The cosine of the argument of periapsis.
* `sin_omega`/`sin_ω` -- The sine of the argument of periapsis.
* `incl` -- The inclination of the orbital plane relative to the axis perpendicular to the
           reference plane [rad]
* `omega`/`ω` -- The argument of periapsis [rad].
* `Omega`/`Ω` -- The longitude of the ascending node [rad].
* `M_planet`/`Mp` -- The mass of the planet [M⊙].
* `M_star`/`Ms` -- The mass of the star [M⊙].
```

my thoughts: I think the subscript/superscript unicode is a lot to type, so we can stick to ~1 unicode character per parameter. This list looks usable and let's unicode still be really concise where it makes sense.

Reply:
I agree that it makes for extra typing, but the way suggested above seems to introduce some inconsistencies into the naming conventions we have, I think. For example, `tp` is now used as a "standard" parameter name while `Rp` is used as an "alternative" one. We also seem to be mixing and matching when `s` is used as a placeholder for `star`. For example in `rho_star`/`ρ_star` vs. `M_star`/`Ms`. Do you want to just go with `s` over `star` for the optional names (e.g. `rho_star`/`ρ_star` -> `rho_star`/`ρs`)?